src/bodse/timetable/GTFS_download.py

```diff
@@ -0,0 +1,107 @@
+import requests
+from datetime import datetime, timedelta
+from pathlib import Path
+import zipfile
+import os
+from clint.textui import progress
+import sys
+
+
+def download_gtfs_feed(directory):
+    current_datetime = datetime.now()
+    current_date_time = current_datetime.strftime("%Y%m%d_%H-%M-%S")
+
+    try:
+        print("Downloading started")
+        url = "https://data.bus-data.dft.gov.uk/timetable/download/gtfs-file/all"
+
+        req = requests.get(url, stream=True)
+
+        filename = Path("GTFS_" + current_date_time + ".zip")
+        filepath = directory / filename
+
+        with open(filepath, "wb") as output_file:
+            total_length = int(req.headers.get("content-length"))
+            for chunk in progress.bar(
+                req.iter_content(chunk_size=1024),
+                expected_size=(total_length / 1024) + 1,
+            ):
+                if chunk:
+                    output_file.write(chunk)
+        print("Downloading Completed")
+
+        # Extract the downloaded zip file to a folder with the same name
+        extraction_folder = (
+            directory / filename.stem
+        )  # Using filename.stem to get the name without the extension
+        extraction_folder.mkdir(exist_ok=True)
+
+        with zipfile.ZipFile(filepath, "r") as zip_ref:
+            zip_ref.extractall(extraction_folder)
+
+        print(f"Extracted contents to {extraction_folder}")
+
+        # Create an "ss" folder and move previous files
+        ss_folder = directory / "ss"
+        ss_folder.mkdir(exist_ok=True)
+
+        for file in directory.glob("GTFS_*"):
+            if file != extraction_folder:
+                new_location = ss_folder / file.name
+                file.rename(new_location)
+                print(f"Moved {file.name} to {new_location}")
+
+    except requests.exceptions.RequestException as e:
+        print("Download failed:", e)
+
+    except zipfile.BadZipFile:
+        print("Failed to unzip the downloaded file.")
+
+    except Exception as e:
+        print("An error occurred:", e)
```

Comment:
Try using `LOG.error` for these with the `exc_info=True` parameter
```suggestion
    except requests.exceptions.RequestException:
        LOG.error("Download failed:", exc_info=True)

    except zipfile.BadZipFile:
        LOG.error("Failed to unzip the downloaded file.", exc_info=True)

    except Exception:
        LOG.error("An error occurred:", exc_info=True)
```